From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Koenn <koenn@bits.team>
Date: Tue, 5 Jan 2021 09:41:47 +0100
Subject: [PATCH] Added ComparatorUpdateEvent


diff --git a/src/main/java/team/bits/internal/event/ComparatorUpdateEvent.java b/src/main/java/team/bits/internal/event/ComparatorUpdateEvent.java
new file mode 100644
index 0000000000000000000000000000000000000000..b6b17b6d9fd0e5783843bd18547155cabb0efd2c
--- /dev/null
+++ b/src/main/java/team/bits/internal/event/ComparatorUpdateEvent.java
@@ -0,0 +1,37 @@
+package team.bits.internal.event;
+
+import org.bukkit.block.Block;
+import org.bukkit.event.HandlerList;
+import org.bukkit.event.block.BlockEvent;
+import org.jetbrains.annotations.NotNull;
+
+import java.util.Objects;
+
+public class ComparatorUpdateEvent extends BlockEvent {
+
+    private static final HandlerList handlers = new HandlerList();
+
+    protected int redstoneOutput;
+
+    public ComparatorUpdateEvent(@NotNull Block block, int redstoneOutput) {
+        super(Objects.requireNonNull(block));
+        this.redstoneOutput = redstoneOutput;
+    }
+
+    public int getRedstoneOutput() {
+        return this.redstoneOutput;
+    }
+
+    public void setRedstoneOutput(int redstoneOutput) {
+        this.redstoneOutput = redstoneOutput;
+    }
+
+    @Override
+    public @NotNull HandlerList getHandlers() {
+        return handlers;
+    }
+
+    public static @NotNull HandlerList getHandlerList() {
+        return handlers;
+    }
+}
